packages/commonwealth/server/routes/tokenBalance.ts

```diff
@@ -45,20 +47,33 @@ const tokenBalance = async (
     throw new AppError(err)
   }
 
+  try {
+    const { contract_address } = req.body;
+    contract = await models.Contract.findOne({
+      where: {
+        address: contract_address,
+      }
+    });
+  } catch (err) {
+    throw new AppError(err);
+  }
+
+
+
   try {
     const balance = await tokenBalanceCache.getBalance(
       chain.chain_node_id,
```

Comment:
How do we get this chain_node_id?

Reply:
from the chain returned at the top of the file from `validateChain()`.

Reply:
What if the contract we're checking uses a different node id?

Reply:
fixed